src/bp-core/bp-core-functions.php

```diff
@@ -4894,7 +4894,7 @@ function bp_get_deprecated_functions_versions() {
 	 */
 	$initial_version        = (float) bp_get_initial_version();
 	$current_version        = (float) bp_get_version();
-	$load_latest_deprecated = $initial_version < $current_version;
+	$load_latest_deprecated = $initial_version < bp_get_major_version( $current_version );
```

Comment:
**`bp_get_major_version( $current_version )`** returns a **string** value, e.g., "12.0", and this line is making a comparison between a float value (**`$initial_version`**) and the aforementioned **string** value.  Is this what you want to do?

Reply:
You are correct; comparing a float value ($initial_version) and a string value ($current_version) is inappropriate and may lead to unexpected behavior or incorrect results.
$current_major_version  = (float) bp_get_major_version( $current_version );
$load_latest_deprecated = $initial_version < $current_major_version;

Reply:
Many thanks to you both, I'll update the PR asap

Reply:
I choose to use an array of float values instead.